ps-modules/Build/Build.psm1

```diff
@@ -0,0 +1,265 @@
+Import-Module "$PSScriptRoot/../../ps-modules/Util" -Force -DisableNameChecking
+
+function Install-FromVcpkg {
+    param(
+        [string] $Package,
+        [string] $Triplet,
+        [string] $VcpkgExe
+    )
+
+    $pkgToInstall = "${Package}:${Triplet}"
+    Write-Message "Installing package: `"$pkgToInstall`""
+    Invoke-Expression "$VcpkgExe install `"$pkgToInstall`" --overlay-triplets=`"custom-triplets`" --overlay-ports=`"custom-ports`""
+}
+
+function Write-ReleaseInfoJson {
+    param(
+        [string] $PackageName,
+        [string] $Version,
+        [string] $PathToJsonFile
+    )
+    $releaseInfo = @{
+        packageName = $PackageName
+        version = $Version
+    }
+    $releaseInfo | ConvertTo-Json | Set-Content -Path $PathToJsonFile
+}
+
+function Get-PackageInfo
+{
+    param(
+        [string]$PackageName
+    )
+    $jsonFilePath = "preconfigured-packages.json"
+    Write-Message "Reading config from: `"$jsonFilePath`""
+    $packagesJson = Get-Content -Raw -Path $jsonFilePath | ConvertFrom-Json
+    $pkg = $packagesJson.packages | Where-Object { $_.name -eq $PackageName }
+    if (-not $pkg) {
+        Write-Message "> Package not found in $jsonFilePath."
+        exit
+    }
+    $IsOnWindowsOS = Get-IsOnWindowsOS
+    $selectedSection = if ($IsOnWindowsOS) { "win" } else { "mac" }
+    return $pkg.$selectedSection
+}
+
+function Initialize-Variables {
```

Comment:
I wish there was a better name for this function because `Variables` could mean anything.

Thinking about this more, maybe the vars that you pass back could be called a "vcpkg environment"?

I'm fine leaving this as-is too.

Reply:
Yeah,  really only went so far as to just extract a method with this.  I could narrow the interface further and make this better with another pass where I init vars locally in the steps that need them.

For example, I think we can get away with just passing along fewer vars to these methods, for instance, and just initing everything else needed for it inside of there:
- Setup-VcPkg( $VcpkgHash )
- Install-Package( $PackageAndFeatures, $LinkType, $BuildType )

Reply:
OK, how about this? 45c439a6e0fb6204884bd2f9584b08fdb7d576d0